docs/infrastructure/microk8s/install.md

```diff
@@ -0,0 +1,274 @@
+   2. Then we can install `tiller`, which will be installed with the same `HELM_VERSION` that you have specified earlier: `helm init --service-account tiller`
+   3. We wait for tiller to be ready: `microk8s.kubectl -n kube-system wait --for=condition=Ready pod -l name=tiller --timeout=300s`
+
+### Codacy chart
+
+1. Create a namespace inside the cluster onto which codacy will be deployed. `microk8s.kubectl create namespace codacy`
+2. The microk8s cluster must know some docker credentials so that the container docker images can be pulled onto the pods. These credentials must belong to the namespace where codacy is installed. You can do so by running: `microk8s.kubectl create secret docker-registry docker-credentials --docker-username=${DOCKER_USERNAME} --docker-password=${DOCKER_PASSWORD} --namespace codacy`. These credentials will be encoded inside the cluster.
+3. For codacy to work, we must set a file that which contains information like how many replicas of a given pod will codacy have, what are the cpu and memory limits for these pods, which secret tokens to use for the filestore, among others.
+   1. Define the following variables
+      1. `CODACY_PROTOCOL` - http or https protocol for your codacy installation.
+      2. `CODACY_URL` - desired URL for your codacy installation.
+      3. `CRYPTO_SECRET` - A secret...
+   2. Get your `microk8s-values.yaml` file ready by running `cat << END_VALUES > microk8s-values.yaml`
+   3. Paste the contents of [the microk8s-values.yaml file](####microk8s-values.yaml) written below.
+   4. Type `END_VALUES` to finish the inline `cat` onto the file.
+4. `helm` must know where it can get the charts from. You must add Codacy's chart museum to `helm`'s internal list: `
+helm repo add codacy-repo https://charts.codacy.com/stable/`
+5. Finally, you can install Codacy's chart by issuing the following `helm` command: `helm upgrade --install codacy codacy-repo/codacy --namespace codacy \
+  --recreate-pods --timeout 1200 --values microk8s-values.yaml`
+   1. This command will tell `helm` to install codacy from scratch if no installation exists. Otherwise it will additively upgrade the version that is installed.
+   2. It will be installed onto the given `--namespace`.
+   3. It will also tell `tiller` to terminate all running pods and create new ones (`--recreate-pods`). This is useful for propagating changes in configuration across all pods.
+   4. `helm` will wait for the given `--timeout` to have all the pods in a `Running` state.
+   5. Lastly we provide the `microk8s-values.yaml` with the intended `--value` for the cluster's configuration.
+
+
+
+
+#### microk8s-values.yaml
+
+```yaml
+global:
+  imagePullSecrets:
+    - name: docker-credentials
+  codacy:
+    url: "${CODACY_PROTOCOL}://${CODACY_URL}"
+    backendUrl: "${CODACY_PROTOCOL}://${CODACY_URL}"
+  akka:
+    sessionSecret: "${CRYPTO_SECRET}"
```

Comment:
I think we should encourage users to use different secrets instead of reusing one.

Reply:
That's a really good point. Changing it!